guides/historical_data.md

```diff
@@ -0,0 +1,94 @@
+# Configuring historical data
+
+If you wish to populate metrics with current historical data saved from telemetry or another data source,
+modify the dashboard config to include a historical_data key like so:
+
+```elixir
+live_dashboard "/dashboard",
+    metrics: MyAppWeb.Telemetry,
+    historical_data: {MyStorage, :historical_metric_data, []}
+```
+
+where MyStorage is a module and historical_metric_data is a function taking a single argument in this example, which will always be a metric.  The function must return a list, empty if there is no data, or a list of maps with `:label`, `:measurement` and `:time` keys in every map.  The measurement should be the output of `Phoenix.LiveDashboard.TelemetryListener.extract_measurement`
+and the label should be the output of `Phoenix.LiveDashboard.TelemetryListener.tags_to_label`, and time should be in `:native` time unit with microsecond precision, such as from `System.system_time(:microsecond)`.
+
+As an example, if you want history for all metrics. You can store history for those metrics, perhaps in a [circular buffer](https://en.wikipedia.org/wiki/Circular_buffer), as in the example below, and emit recent telemetry when each client connects and LiveDashboard calls into your module for historical data for the metrics on that tab.  If using this example you would also need to and add the module to your Application children, and initialize it with some or all of your metrics, such as from `MyAppWeb.Telemetry.metrics/0` . You could also store the data in an ETS table or in Redis or the database, or anywhere else, but for this example we'll show using a GenServer:
+
+```elixir
+  defmodule DemoWeb.History do
+    use GenServer
+    alias Phoenix.LiveDashboard.TelemetryListener
+
+    @history_buffer_size 50
+
+    def data(metric) do
+      GenServer.call(__MODULE__, {:data, metric})
+    end
```

Comment:
This and the callback seem to be unused code.

Reply:
This is the hook that would be passed into historical_data hook and would return all data, unless I'm confused what you mean.

Reply:
Doh, I missed that, thanks!